packages/compiler/src/core/helpers/type-name-utils.ts

```diff
@@ -1,26 +1,31 @@
 import { isDefined } from "../../utils/misc.js";
+import { Checker } from "../checker.js";
 import { isTemplateInstance, isType, isValue } from "../type-utils.js";
-import type {
-  Entity,
-  Enum,
-  Interface,
-  Model,
-  ModelProperty,
-  ModelStatementNode,
-  Namespace,
-  Operation,
-  Scalar,
-  StringTemplate,
-  Type,
-  Union,
-  Value,
+import {
+  Expression,
+  type Entity,
+  type Enum,
+  type Interface,
+  type Model,
+  type ModelProperty,
+  type ModelStatementNode,
+  type Namespace,
+  type Operation,
+  type Scalar,
+  type StringTemplate,
+  type Type,
+  type Union,
+  type Value,
 } from "../types.js";
 import { printIdentifier } from "./syntax-utils.js";
 
+const expressionDefaultCache = new Map<Expression, string>();
+
 export interface TypeNameOptions {
   namespaceFilter?: (ns: Namespace) => boolean;
   printable?: boolean;
   nameOnly?: boolean;
+  checker?: Checker;
```

Comment:
I dont think it's a good idea to put checker here as an "option"

Reply:
+1 on this, this is definitely wrong